txpipe/photoz_dir.py

```diff
@@ -0,0 +1,189 @@
+    }
+
+    def run(self):
+        """
+        Run the analysis for this stage.
+        
+         - Get metadata and allocate space for output
+         - Set up iterators to loop through tomography and PDF input files
+         - Accumulate the PDFs for each object in each bin
+         - Divide by the counts to get the stacked PDF
+        """
+
+        logger.info("Getting COSMOS N(z)s.")
+        pzs_cosmos = self.get_nz_cosmos('source')
+
+        # Save the stacks
+        f = self.open_output("shear_photoz_dir")
+        self.save('source', pzs_cosmos, f)
+        f.close()
+
+    def get_nz_cosmos(self, name):
+        """
+        Get N(z) from weighted COSMOS-30band data
+        """
+
+        if name == 'source':
+            zi_arr = self.config['source_zbin_edges'][:-1]
+            zf_arr = self.config['source_zbin_edges'][1:]
+        elif name == 'lens':
+            zi_arr = self.config['lens_zbin_edges'][:-1]
+            zf_arr = self.config['lens_zbin_edges'][1:]
+        else:
+            raise NotImplementedError('Only name = source, lens supported. Aborting.')
+
+        if self.config['pz_code'] == 'ephor_ab':
+            pz_code = 'eab'
+        elif self.config['pz_code'] == 'frankenz':
+            pz_code = 'frz'
+        elif self.config['pz_code'] == 'nnpz':
+            pz_code = 'nnz'
+        else:
+            raise KeyError("Photo-z method "+self.config['pz_code'] +
+                           " unavailable. Choose ephor_ab, frankenz or nnpz")
+
+        if self.config['pz_mark'] not in ['best', 'mean', 'mode', 'mc']:
+            raise KeyError("Photo-z mark "+self.config['pz_mark'] +
+                           " unavailable. Choose between best, mean, " +
+                           "mode and mc")
+
+        self.column_mark = 'pz_'+self.config['pz_mark']+'_'+pz_code
+
+        if name == 'source':
+            weights_file = fits.open(self.get_input('cosmos_source_weights'))[1].data
+        elif name == 'lens':
+            weights_file = fits.open(self.get_input('cosmos_photo_weights'))[1].data
+        else:
+            raise NotImplementedError('Only name = source, lens supported. Aborting.')
+
+        pzs = []
+        for zi, zf in zip(zi_arr, zf_arr):
+            msk_cosmos = ((weights_file[self.column_mark] <= zf) &
+                          (weights_file[self.column_mark] > zi))
+            hz, bz = np.histogram(weights_file[msk_cosmos]['PHOTOZ'],
+                                  bins=self.config['nz_bin_num'],
+                                  range=[0., self.config['nz_bin_max']],
+                                  weights=weights_file[msk_cosmos]['weight'])
+            hnz, bnz = np.histogram(weights_file[msk_cosmos]['PHOTOZ'],
+                                    bins=self.config['nz_bin_num'],
+                                    range=[0., self.config['nz_bin_max']])
+            ehz = np.zeros(len(hnz))
+            ehz[hnz > 0] = (hz[hnz > 0]+0.)/np.sqrt(hnz[hnz > 0]+0.)
```

Comment:
`hz` is a float, I don't think that you need to add `0.` (Also I think that we are not supporting Python2x anyway)